protocol/x/ratelimit/keeper/keeper_test.go

```diff
@@ -51,72 +58,310 @@ func TestSetGetDenomCapacity(t *testing.T) {
 }
 
 func TestSetGetLimitParams_Success(t *testing.T) {
-	tApp := testapp.NewTestAppBuilder(t).Build()
-	ctx := tApp.InitChain()
-	k := tApp.App.RatelimitKeeper
-
-	limiters := []types.Limiter{
-		{
-			PeriodSec:       3_600,
-			BaselineMinimum: dtypes.NewInt(100_000_000_000), // 100k tokens (assuming 6 decimals)
-			BaselineTvlPpm:  10_000,                         // 1%
+	// Run tests.
+	tests := map[string]struct {
+		denom                string
+		balances             []banktypes.Balance
+		limiters             []types.Limiter
+		expectedCapacityList []dtypes.SerializableInt
+	}{
+		"0 TVL, capactiy correctly initialized as minimum baseline": {
+			denom: testDenom,
+			limiters: []types.Limiter{
+				{
+					PeriodSec:       3_600,
+					BaselineMinimum: dtypes.NewInt(100_000_000_000), // 100k tokens (assuming 6 decimals)
+					BaselineTvlPpm:  10_000,                         // 1%
+				},
+				{
+					PeriodSec:       86_400,
+					BaselineMinimum: dtypes.NewInt(1_000_000_000_000), // 1m tokens (assuming 6 decimals)
+					BaselineTvlPpm:  100_000,                          // 10%
+				},
+			},
+			expectedCapacityList: []dtypes.SerializableInt{
```

Comment:
for my own understanding, will there be capacity per denom?

Reply:
This list is per denom. See test body below - 
```
			// Expected `DenomCapacity` is initialized such that each capacity is equal to the baseline.
			expectedDenomCapacity := types.DenomCapacity{
				Denom:        tc.denom,
				CapacityList: tc.expectedCapacityList,
			}
```